src/apps/spool-v2/ethereum/spool-v2.staking.contract-position-fetcher.ts

```diff
@@ -0,0 +1,154 @@
+          .map(reward => reward.token.id)
+          .concat([SPOOL_ADDRESS]),
+      );
+    }
+
+    return [
+      { metaType: MetaType.SUPPLIED, address: SPOOL_ADDRESS, network: this.network },
+      { metaType: MetaType.VESTING, address: VOSPOOL_ADDRESS, network: this.network },
+      ...rewardAddresses.map(address => ({ metaType: MetaType.CLAIMABLE, address, network: this.network })),
+    ];
+  }
+
+  async getDataProps({ multicall, contractPosition }: GetDataPropsParams<SpoolStaking, SpoolStakingDataProps>) {
+    const [spoolToken, voSpoolToken] = contractPosition.tokens;
+    const spoolContract = this.appToolkit.globalViemContracts.erc20(spoolToken);
+    const voSpoolContract = this.contractFactory.spoolVospool(voSpoolToken);
+
+    const totalStakedRaw = await multicall.wrap(spoolContract).read.balanceOf([STAKING_ADDRESS]);
+    const votingPowerRaw = await multicall.wrap(voSpoolContract).read.getTotalGradualVotingPower();
+
+    const totalAccVoSpool = Number(votingPowerRaw) / 10 ** voSpoolToken.decimals;
+    const spoolStaked = Number(totalStakedRaw) / 10 ** spoolToken.decimals;
+    const tvl = spoolStaked * spoolToken.price;
+
+    return { tvl, spoolStaked, totalAccVoSpool };
+  }
+
+  async getLabel({ contractPosition }: GetDisplayPropsParams<SpoolStaking>) {
+    return `${getLabelFromToken(contractPosition.tokens[0])} Staking`;
+  }
+
+  async getStatsItems({ contractPosition }: GetDisplayPropsParams<SpoolStaking, SpoolStakingDataProps>) {
+    return [
+      { label: 'TVL', value: buildDollarDisplayItem(contractPosition.dataProps.tvl) },
+      { label: 'SPOOL Staked', value: buildNumberDisplayItem(contractPosition.dataProps.spoolStaked) },
+      { label: 'voSPOOL Accumulated', value: buildNumberDisplayItem(contractPosition.dataProps.totalAccVoSpool) },
+    ];
+  }
+
+  async getTokenBalancesPerPosition({
+    address,
+    contractPosition,
+    multicall,
+  }: GetTokenBalancesParams<SpoolStaking, SpoolStakingDataProps>) {
+    const suppliedToken = contractPosition.tokens.find(isSupplied)!;
+    const rewardTokens = contractPosition.tokens.filter(isClaimable);
+
+    const staking = this.contractFactory.spoolStaking({ address: STAKING_ADDRESS, network: this.network });
+    const voSpool = this.contractFactory.spoolVospool({ address: VOSPOOL_ADDRESS, network: this.network });
+
+    const [votingPowerRaw, voSpoolRewards, ...tokenRewards] = await Promise.all([
+      multicall.wrap(voSpool).read.getUserGradualVotingPower([address]),
+      multicall
+        .wrap(staking)
+        .simulate.getUpdatedVoSpoolRewardAmount({ account: address })
+        .then(v => v.result),
+      ...rewardTokens.map(reward => multicall.wrap(staking).read.earned([reward.address, address])),
+    ]);
+
+    const stakedSpool = await gqlFetch<UserSpoolStaking>({
+      endpoint: SUBGRAPH_V1_API_BASE_URL,
+      query: SPOOL_STAKED_QUERY,
+      variables: { address },
+    });
```

Comment:
Look like you can call `balanceOf` on `0xc3160c5cc63b6116dd182faa8393d3ad9313e213` to get staked balance